TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -134,15 +141,150 @@ public async Task<CommandResult> SelectBadge(CommandContext context)
+                        ResponseTarget = ResponseTarget.WhisperIfLong
+                    };
+                }
+                else if( mode.Equals( PokedexModeDupes ) )
+                {
+                    var dupeList = numBadgesPerSpecies.Where(kvp => kvp.Value > 1);
+                    if( !dupeList.Any() )
+                    {
+                        return new CommandResult
+                        {
+                            Response = isSelf
+                                ? $"You do not own any duplicate Pokémon badges"
+                                : $"{user.Name} does not own any duplicate Pokémon badges"
+                        };
+                    }
+                    else
+                    {
+                        IEnumerable<string> badgesFormatted = dupeList.Select( kvp => $"{kvp.Key}" );
+                        return new CommandResult
+                        {
+                            Response = isSelf
+                                ? $"You are owning duplicates of the following badge(s): {string.Join(", ", badgesFormatted)}"
+                                : $"{user.Name} is owning duplicates of the following badge(s): {string.Join(", ", badgesFormatted)}",
+                            ResponseTarget = ResponseTarget.WhisperIfLong
+                        };
+                    }
+                }
+                else if( mode.Equals( PokedexModeComplementFromDupes ) )
+                {
+                    if( !optionalUser2.IsPresent )
+                    {
+                        return new CommandResult
+                        {
+                            Response = $"Mode {PokedexModeComplementFromDupes} requires a second user argument"
+                        };
+                    }
+                    else
+                    {
+                        var dupeList = numBadgesPerSpecies.Where( kvp => kvp.Value > 1 );
+                        ImmutableSortedDictionary<PkmnSpecies, int> myNumBadgesPerSpecies =
+                            await _badgeRepo.CountByUserPerSpecies(optionalUser2.Value.Id);
+
+                        var intersectList = dupeList.Except( myNumBadgesPerSpecies );
+                        if( !intersectList.Any() )
+                        {
+                            return new CommandResult
+                            {
+                                Response = $"{user.Name} does not own any duplicate Pokémon badges {optionalUser2.Value.Name} is missing"
+                            };
+                        }
+                        else
+                        {
+                            IEnumerable<string> badgesFormatted = intersectList.Select( kvp => $"{kvp.Key}" );
+                            return new CommandResult
+                            {
+                                Response = $"{user.Name} is owning missing duplicates of the following badge(s): {string.Join(", ", badgesFormatted)}",
```

Comment:
message is missing whose user the dupes are for

Reply:
I'll rework that code. Output both users first and then list the result